src/gui/screen_print_preview.cpp

```diff
@@ -19,37 +19,97 @@ const uint16_t menu_icons[2] = {
+        // this MakeRAM is safe - vars->media_LFN is statically allocated (even though it may not be obvious at the first look)
+#if defined(USE_ST7789)
+        switch (MsgBoxTitle(string_view_utf8::MakeRAM((const uint8_t *)gcode.GetGcodeFilename()), txt_fil_not_detected, btns, 0, GuiDefaults::DialogFrameRect)) {
+#else
+    #error "Unknown display"
+#endif                      // USE_<DISPLAY>
+        case Response::Yes: //YES - load
+            PreheatStatus::DialogBlocking(PreheatMode::Load, RetAndCool_t::Return);
             break;
-        case Response::Cancel:
-            Sound_Play(eSOUND_TYPE::SingleBeep);
-            approved = false;
+        case Response::No: //NO - cancel
+            return false;
+        case Response::Ignore: //IGNORE - disable
+            FS_instance().Disable();
+            return true;
+        default:
             break;
+        }
+    }
+    return true;
+}
+
+/// \returns true if filament has (finally) the correct type or the type is ignored
+static bool check_filament_type(GCodeInfo &gcode) {
+    const char *curr_filament = Filaments::Current().name;
+    while (gcode.filament_described && strncmp(curr_filament, "---", 3) != 0 && strncmp(curr_filament, gcode.filament_type, sizeof(gcode.filament_type)) != 0) {
+        string_view_utf8 txt_wrong_fil_type = _("This G-CODE was set up for another filament type.");
+#if defined(USE_ST7789)
+        switch (MsgBoxWarning(txt_wrong_fil_type, Responses_ChangeIgnoreAbort, 0)) {
+#else
+    #error "Unknown display"
+#endif // USE_<DISPLAY>
+        case Response::Change:
+            PreheatStatus::DialogBlocking(PreheatMode::Change_phase1, RetAndCool_t::Return);
+            break;
+        case Response::Ignore:
+            return true;
+        case Response::Abort:
+            return false;
         default:
             break;
         }
     }
-    if (approved) {
-        print_begin(gcode.GetGcodeFilepath());
+    return true;
+}
+
+/// \returns true if it's correct printer or printer type is ignored
+static bool check_printer_type(GCodeInfo &gcode) {
+    if (gcode.IsSettingsValid())
+        return true;
+    string_view_utf8 txt_wrong_printer_type = _("This G-CODE was set up for another printer type.");
+#if defined(USE_ST7789)
+    switch (MsgBoxWarning(txt_wrong_printer_type, Responses_IgnoreAbort, 0)) {
+#else
+    #error "Unknown display"
+#endif // USE_<DISPLAY>
+    case Response::Abort:
+        return false;
+    default:
+        break;
+    }
+    return true;
+}
+
+static void print_button_pressed() {
+    GCodeInfo &gcode = GCodeInfo::getInstance();
+    if (!check_printer_type(gcode)
+        || !check_filament_presence(gcode)
+        || !check_filament_type(gcode)) {
+        Sound_Play(eSOUND_TYPE::SingleBeep);
+        Screens::Access()->Close();
+        return;
```

Comment:
Do we want to check the if the print is possible when user clicks on the prin button?